File: Parallel.java

public class Parallel {
	public static void main(String args[]) throws InterruptedException {
		Runnable task1 = () -> {
			try {
				Thread.sleep(1000);
			}
			catch(InterruptedException e) {
				System.out.println("Exception ::"+e.getMessage());
			}
		};
		
		Runnable task2 = () -> {
			try {
				Thread.sleep(2000);
			}
			catch(InterruptedException e) {
				System.out.println("Exception ::"+e.getMessage());
			}
		};
		
		
		Runnable task3 = () -> {
			try {
				Thread.sleep(3000);
			}
			catch(InterruptedException e) {
				System.out.println("Exception ::"+e.getMessage());
			}
		};
		
		Runnable task4 = () -> {
			try {
					Thread.sleep(4000);
				}
			catch(InterruptedException e) {
				System.out.println("Exception ::"+e.getMessage());
			}
		};
		
		long startTime = System.currentTimeMillis();
		
		System.out.println("Start!");
		Thread thread1 = new Thread(task1);
		thread1.start();
		Thread thread2 = new Thread(task2);
		thread2.start();
		Thread thread3 = new Thread(task3);
		thread3.start();
		Thread thread4 = new Thread(task4);
		thread4.start();
		thread4.join();
		System.out.println("Done!");
		
		long endTime = System.currentTimeMillis();
		System.out.println("Time taken: "+((endTime- startTime)/1000 )+" sec" );
		
	}
	
}

File: parallel.py
import threading
import time
print("Start")
def Testcase1():
    time.sleep(1)
def Testcase2():
    time.sleep(2)
def Testcase3():
    time.sleep(3)
def Testcase4():
    time.sleep(4)
start_time = time.time()
threading.Thread(target=Testcase1).start()
threading.Thread(target=Testcase2).start()
threading.Thread(target=Testcase3).start()
#threading.Thread(target=Testcase4()).start()
t1 = threading.Thread(target=Testcase4())
t1.start()
t1.join()
end_time = time.time()
print("total time for 4 testcases ",round(end_time-start_time), "seconds")
print("Done")

